htdocs/core/tpl/extrafields_list_search_sql.tpl.php

```diff
@@ -35,7 +35,8 @@
 		{
 			$mode_search = 0;
 			if (in_array($typ, array('int', 'double', 'real', 'price'))) $mode_search = 1; // Search on a numeric
-			if (in_array($typ, array('sellist', 'link')) && $crit != '0' && $crit != '-1') $mode_search = 2; // Search on a foreign key int
+			if (in_array($typ, array('sellist', 'link')) && $crit != '0' && $crit != '-1' && is_numeric($crit)) $mode_search = 2; // Search on a foreign key int
```

Comment:
But what if $typ = 'link' and $crit is not numeric, there is no case here where there was before.
I think old line must keep as it was and you add just an exception to overwrite mode_search if selllist and crit is not a numeric.

Reply:
doone